system/core/database.php

```diff
@@ -599,10 +599,40 @@ function __construct($filename) {
 		function lock() {
 			if ($this->fd)
```

Comment:
Terzijde opmerking: de naamte `fd` maakte het voor mij niet direct duidelijk waar dit om ging, dat kan wellicht verduidelijkt worden :)

Reply:
fd slaat op file descriptor, wat iig in C-code wel een veel gebruikte naam is. Ik zie dat de fd variable niet gedeclareerd is, dus dat kan ik wel even doen met een commentje erbij.

Reply:
Oh wacht, ik zat bij `HyphaImage` te kijken, in `HyphaFile` is `$fd` al wel gedeclareerd is met een commentje (wel met foutje, null vs false). Blijkbaar is dat nog niet voldoende om het duidelijk te maken :-p

Kijkende naar [de docs](https://www.php.net/manual/en/function.fopen.php) is het eigenlijk ook geen file descriptor (dat zijn ints, e.g. 0 voor stdin, 1 voor stdout, 2 voor stderr en 3+ voor opened files) die teruggeven wordt, maar een of ander PHP-specifiek resource object, dus is een andere naam misschien wel een goed idee.

Ik denk aan `$file_handle`, zou dat voor jouw duidelijker zijn geweest?

Reply:
`$file_handle` = top :)

Reply:
Done.